src/main/scala/tech/sourced/gemini/ReportApp.scala

```diff
@@ -166,4 +201,32 @@ object ReportApp extends App {
     writerBuckets.close()
   }
 
+  def readCommunities(dirPath: String): List[(Int, List[Int])] = {
+    val parquetFilePath = new Path(s"$dirPath/communities.parquet")
+    val reader = AvroParquetReader.builder[GenericRecord](parquetFilePath).build()
+
+    var result = mutable.Map[Int, List[Int]]()
+
+    var nextRecord: GenericRecord = null
+
+    while ( {
+      nextRecord = reader.read
+      nextRecord
+    } != null) {
```

Comment:
Wow, this looks like written in C.

Could you find any other options to re-write this iteration construct?

Reply:
I tried to in c24eab1, other suggestions are welcome.